### PR TITLE
[Routing] Keep declared routes in shared memory when possible on PHP 5.6

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Branch? | 2.3 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

**Comment by javiereguiluz** [comment]
@nicolas-grekas thanks for yet another great contribution!

**Comment by stof** [comment]
we should have tests covering this

**Comment by nicolas-grekas** [comment]
This is already feature tested. Having a unit test is really hard for the side effect that is achieved here.

**Comment by xabbuh** [comment]
@nicolas-grekas Do you have a profile showing differences compared with the previous code?

**Comment by nicolas-grekas** [comment]
@xabbuh no I don't, but I confirmed the behavior described in the comment with the following scripts (or variants of):

```php
<?php
$s = memory_get_peak_usage();
$range = require './static.php';
$f = new toto;
$e = memory_get_peak_usage();
echo $e - $s, "\n";
```

```php
<?php class toto {
private $range;
function __construct($x = 1) {if ($x) {static $range = array (
  0 => 1,
//...
);}
$this->range =& $range;
unset($range);
}}
```
running into a php -S with opcache enabled.

File: src/Symfony/Component/Routing/Generator/Dumper/PhpGeneratorDumper.php [hunk 1]
```diff
@@ -38,6 +38,11 @@ public function dump(array $options = array())
             'base_class' => 'Symfony\\Component\\Routing\\Generator\\UrlGenerator',
         ), $options);
 
+        // On PHP 5.6, OPcache is able to keep statically declared arrays in shared memory,
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Routing/Generator/Dumper/PhpGeneratorDumper.php]: I can't verify this statement, :(

**Pull request closed without merging**